pkg/secrethub/client.go

```diff
@@ -110,7 +115,19 @@ func NewClient(with ...ClientOption) (*Client, error) {
 
 	// Try to use default key credentials if none provided explicitly
 	if client.decrypter == nil {
-		err := client.with(WithCredentials(credentials.UseKey(client.DefaultCredential())))
+		identityProvider := os.Getenv("SECRETHUB_IDENTITY_PROVIDER")
+
+		var provider credentials.Provider
+		switch identityProvider {
+		case "key", "":
+			provider = credentials.UseKey(client.DefaultCredential())
+		case "aws":
+			provider = credentials.UseAWS()
+		default:
+			return nil, ErrUnknownIdentityProvider(identityProvider)
```

Comment:
Should it error here or just ignore it and use the default (a key) instead?

Reply:
If the identity provider is explicitly set to an invalid value, I think it's best to error.

For example, when making a typo in the envvar, setting e.g. `SECRETHUB_IDENTITY_PROVIDER=awss`, I think it would be better to error and explain that `awss` is not a valid value than to continue and either:
- Find a credential at `~/.secrethub/credential` and authenticate as another user (which could result in permission errors)
- Not find a credential and make unauthenticated requests (which will return an error on routes that require authentication)
I think in both cases, the returned errors don't explain the issue (typo in envvar) clearly.